staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go

```diff
@@ -159,8 +182,13 @@ func (s *Serializer) Decode(originalData []byte, gvk *schema.GroupVersionKind, i
 		return versioned, actual, nil
 	}
 
+	var errorYAMLToJSON error
 	data := originalData
 	if s.yaml {
+		// Check for strict YAML to JSON conversation errors.
+		if s.strict {
+			_, errorYAMLToJSON = yaml.YAMLToJSONStrict(data)
```

Comment:
Add comment here that error handling is delayed by design, otherwise this seems very odd

Reply:
Also comment that this call errors only if (right?) there are duplicate fields, and hence it's put here, before the data variable is mutated with duplicate fields removed from non-strict parsing. That'd clearify things a lot for the reader.